pkg/oauthflow/pkce.go

```diff
@@ -57,7 +58,7 @@ func NewPKCE(provider *oidc.Provider) (*PKCE, error) {
 			fmt.Printf("Unsupported code challenge method in list: '%v'", method)
 		}
 	}
-	if chosenMethod == "" {
+	if chosenMethod == "" && !providerIsAzureBacked(provider) {
```

Comment:
This means we pass an empty string as the `code_challenge_method` query parameter value which is wrong. This needs to be set to `S256` as per RFC7636 it should support it.